app/controllers/users_controller.rb

```diff
@@ -0,0 +1,29 @@
+class UsersController < ApplicationController
+  before_action :require_logout!, except: [:show]
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      session[:user_id] = @user.id
+      flash[:notice] = 'User was successfully created'
+      redirect_to user_path(@user)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
```

Comment:
- You do not necessarily have to find the user because you can easily use the `current_user` as the user in your view.
- Also, you should fetch the current user events from this action. i.e. `@created_events = current_user.events`. Therefore, you can use the `@created_events` directly in your view, instead of using `<% for event in @user.events %>` as you did in `app/views/users/show.html.erb` file.

Reply:
"You do not necessarily have to find the user because you can easily use the current_user as the user in your view."

`UsersController#show` method shows the detail page of the user with the `id` in url `params` and not the detail page of the `current_user`. Kindly let me know if this is not the expected behaviour.

On the second bullet point you are saying I should make a variable in `#show` controller called `@created_events = current_user.events`, it feels like I am adding an extra unnecessary line in my code, I mean I am also going to loop through `@created_events` like `<% for event in @ created_events %>` in the `app/views/users/show.html.erb` file.

Kindly clarify on this points